python/pyspark/sql/session.py

```diff
@@ -473,14 +473,22 @@ def getOrCreate(self) -> "SparkSession":
 
                 url = opts.get("spark.remote", os.environ.get("SPARK_REMOTE"))
 
+                if url is None:
```

Comment:
to confirm: this branch (`if is_remote_only():`) means `--remote` is specified?

Reply:
No, it's for `pyspark-client` library that doesn't contain `pyspark.core` package.
If the package is not available, `is_remote_only()` will return `True`.

Without the change back, it will fail with an assertion error in the `RemoteSparkSession`.